koko_hardware:
    ### Joint configuration
    joint_names:
        - base_roll_joint
        - shoulder_lift_joint
        - shoulder_roll_joint
        - elbow_lift_joint
        - elbow_roll_joint
        - wrist_lift_joint
        - wrist_roll_joint
    endlink: wrist_roll_link
    differential_pairs: [1, 2, 3, 4, 5, 6]
    gear_ratios: [7.1875, 7.1875, 8.2444852941, 7.1875, 8.24448529412, 7.1875, 8.24448529412]
    joint_torque_directions: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
    # id_torque_gains: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
    id_torque_gains: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
    softstop_min_angles: [-2.2, -2.25, -2.6, -2.25, -2.6, -2.25, -2.6]
    softstop_max_angles: [3.3, 0.5, 2.6, .5, 2.6, 0.5, 2.6]
    softstop_tolerance: 0.05
    softstop_torque_limit: 45.0

    ### Motor configuration
    motor_names:
        - base_roll_motor
        - right_motor1
        - left_motor1
        - right_motor2
        - left_motor2
        - right_motor3
        - left_motor3
    motor_ids: [15, 20, 18, 12, 11, 19, 21]
    # motor_current_limits: [1.0, 1.0, 1.0, 1.0, 1.0, 0.5, 0.5]
    motor_current_limits: [0.3, 0.3, 0.3, 0.3, 0.3, 0.1, 0.1]
    current_to_torque_ratios: [0.626, 0.626, 0.626, 0.626, 0.626, 1.0, 1.0]

    ### Calibration configuration
    hardstop_start_angles: [-0.785398, -2.19, -1.570796, 0.0, 1.570796, -0.23, 0.0]
    calibration_initial_angles: [-0.62, 0.99, 0.0, -0.83, 0.0, 0.0, 0.0]
